test/jb_splitsStore/set.test.js

```diff
@@ -0,0 +1,293 @@
+    let mockOperatorStore = await deployMockContract(deployer, jbOperatorStoreAbi);
+    let mockProjects = await deployMockContract(deployer, jbProjectsAbi);
+    let mockDirectory = await deployMockContract(deployer, jbDirectoryAbi);
+
+    // Override auth: by default projectOwner is controller + owner + has permission on SET_SPLITS
+    await mockOperatorStore.mock.hasPermission
+      .withArgs(projectOwner.address, projectOwner.address, PROJECT_ID, SET_SPLITS)
+      .returns(true);
+
+    await mockProjects.mock.ownerOf
+      .withArgs(PROJECT_ID)
+      .returns(projectOwner.address);
+
+    await mockDirectory.mock.controllerOf
+      .withArgs(PROJECT_ID)
+      .returns(projectOwner.address);
+
+    let jbSplitsStoreFact = await ethers.getContractFactory('JBSplitsStore');
+    let jbSplitsStore = await jbSplitsStoreFact.deploy(
+      /*operator=*/mockOperatorStore.address,
+      /*projects=*/mockProjects.address,
+      /*directory=*/mockDirectory.address
+    );
+
+    let splits = createSplitArray(addrs[0].address, 4);
+
+    return { deployer, projectOwner, addrs, jbSplitsStore, splits, mockOperatorStore, mockProjects, mockDirectory };
+  }
+
+  // Create array of JBSplit struct
+  function createSplitArray(_benefAdress, n) {
+    let splits = []
+    for (let i = 0; i < n; i++) {
+      splits.push({
+        preferClaimed: false,
+        percent: Math.floor(10000000 / n),
+        lockedUntil: 0,
+        beneficiary: _benefAdress,
+        allocator: ZERO_ADDRESS,
+        projectId: 0
+      });
+    }
+    return splits;
+  }
+
+  // Expected behavior when no preexisting splits for this group - happy path
+  it('set() and corresponding events', async function () {
+
+    const { projectOwner, addrs, jbSplitsStore, splits, mockOperatorStore, mockDirectory } = await setup();
+
+    // set(..) called from project owner address (which is set in setup() )
+    await mockOperatorStore.mock.hasPermission.returns(false);
+    await mockDirectory.mock.controllerOf.returns(addrs[0].address);
+
+    const tx = await jbSplitsStore.connect(projectOwner).set(
+      /*projectId=*/PROJECT_ID,
+      /*domain=*/DOMAIN,
+      /*group=*/GROUP,
+      /*splits[]=*/splits
+    );
+
+    // Expect one event per split in splits[]
+    await expect(tx)
+      .to.emit(jbSplitsStore, 'SetSplit')
+      .and.to.emit(jbSplitsStore, 'SetSplit')
+      .and.to.emit(jbSplitsStore, 'SetSplit')
+      .and.to.emit(jbSplitsStore, 'SetSplit')
+
+    // Get the current splits (for this proj/dom/group)
+    let splitsStored = await jbSplitsStore.splitsOf(PROJECT_ID, DOMAIN, GROUP);
+
+    for (let idx = 0; idx < splits.length; idx++) {
+      for (let split_key of Object.keys(splits[idx])) {
+        expect(splitsStored[idx].split_key).to.equal(splits[idx].split_key);
+      }
+    }
+  })
+
+  // New splits[] when preexisting ones already exist without any lockedUntil
+  it('New splits', async function () {
```

Comment:
'New splits' is not a very descriptive name for this test. The test should not need a comment above and name should be descriptive enough. Please use JBDirectory tests as a reference.